lib/glob-assets.js

```diff
@@ -0,0 +1,45 @@
+const {basename} = require('path');
+const {isPlainObject, castArray, uniqWith} = require('lodash');
+const pReduce = require('p-reduce');
+const globby = require('globby');
+
+module.exports = async assets =>
+  uniqWith(
+    (await pReduce(
+      assets,
+      async (result, asset) => {
+        // Wrap single glob definition in Array
+        const glob = castArray(isPlainObject(asset) ? asset.path : asset);
+        // Skip solo negated pattern (avoid to include every non js file with `!**/*.js`)
+        if (glob.length > 1 || !glob[0].startsWith('!')) {
+          const globbed = await globby(glob, {expandDirectories: true, gitignore: false, dot: true});
+          if (isPlainObject(asset)) {
+            if (globbed.length > 1) {
+              // If asset is an Object with a glob the `path` property that resolve to multiple files,
+              // Output an Object definition for each file matched and set each one with:
+              // - `path` of the matched file
+              // - `name` based on the actual file name (to avoid assets with duplicate `name`)
+              // - other properties of the original asset definition
+              result.push(...globbed.map(file => Object.assign({}, asset, {path: file, name: basename(file)})));
+            } else {
```

Comment:
I would suggest to avoid all the `else` blocks, instead return add `return result` e.g. after line 23. It will make the code more readable.

Reply:
We have to push to `result` before returning.

A version without the `if` `else` could be like that https://gist.github.com/pvdlg/b4764bbf747fc36db8e8846e7912abc1
Does that looks better ?

Reply:
yes. Maybe we can negate the first if condition to return early and avoid an indetation level

```js
if (glob.length <= 1 || glob[0].startsWith('!')) {
  return
}
```

Reply:
Good suggestion. I made both changes.